pkg/events/derive/net_flow.go

```diff
@@ -0,0 +1,165 @@
+package derive
+
+import (
+	"github.com/aquasecurity/tracee/pkg/dnscache"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+func NetFlowTCPBegin(cache *dnscache.DNSCache) DeriveFunction {
+	return deriveSingleEvent(events.NetFlowTCPBegin,
+		func(event trace.Event) ([]interface{}, error) {
+			tcpBegin := event.ReturnValue&flowTCPBegin == flowTCPBegin
+			ingress := event.ReturnValue&packetIngress == packetIngress
+			egress := event.ReturnValue&packetEgress == packetEgress
+
+			// Sanity check
+			if (!ingress && !egress) || (ingress && egress) {
+				logger.Debugw("wrong flow direction", "id", event.EventID)
```

Comment:
I wanted to have a way to log if packets without ingress or egress would ever arrive here (corrupted perhaps ?) but without an explicit error (not sure if under high loads we would cause the network packet events corruption very often). Preferred to be cautious.